### ISSUE TITLE
Rule Change: `logical-assignment-operators` not checking `a = a || 1 || 2`

### ISSUE THREAD

**Issue by fisker** [eslint/eslint#17096]
### What rule do you want to change?

logical-assignment-operators

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

Other

### Example code

```js
a = a || 1
a = a || 1 || 2
a = a || (1 || 2)
```

### What does the rule currently do for this code?

```js
a = a || 1 // error
a = a || 1 || 2 // no error
a = a || (1 || 2) // error
```

### What will the rule do after it's changed?

Both error

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
> ```js
> a = a || 1 || 2 // no error
> ```

This is because `||` is left-associative so this parses as `a = (a || 1) || 2` and therefore the rule does not see it as `a = a || something`.

I think we can update the rule to report `a = a || 1 || 2` and treat this change as a semver-minor bug fix. Curious what other team members think.

**Comment by fisker** [comment]
Once we decide to accept, If someone wants to fix it, [this piece of code maybe useful](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/a9337e847e0ca4d725bdedf8f13e24c7d0dfcda9/rules/no-useless-length-check.js#L35)

**Comment by mdjermanovic** [comment]
I'm just not sure if the same should apply when the left-hand side of `||` is in parentheses because perhaps the intention was to visually separate that expression for some reason. For example,  `a = (a || 1) || 2`. This probably still shouldn't be reported even though it's the same as `a = a || 1 || 2`.

**Comment by Rec0iL99** [comment]
> > ```js
> > a = a || 1 || 2 // no error
> > ```
> 
> This is because `||` is left-associative so this parses as `a = (a || 1) || 2` and therefore the rule does not see it as `a = a || something`.
> 
> I think we can update the rule to report `a = a || 1 || 2` and treat this change as a semver-minor bug fix. Curious what other team members think.

Not stale. Waiting for feedback.

cc @eslint/eslint-team

**Comment by B2o5T** [comment]
for me, it's a bug, I already used mentioned @fisker's function `flatLogicalExpression` that makes work with logical expressions very easy

**Comment by aladdin-add** [comment]
```js
a = a || b || c;
```
for the case, it can be safely rewritten to `a ||= b || c`??? 😔

**Comment by fisker** [comment]
@aladdin-add Is there case will be broken?

**Comment by aladdin-add** [comment]
there is a difference:
```js
a = a || b || c; // same as: a = (a || b) || c;

a ||= b || c; // same as: a = a || (b || c);
```

But after I thought about it, they should have the same execution result.  @fisker what you mean by "setter will not be called"?

**Comment by fisker** [comment]
>what you mean by "setter will not be called"?

```js
const a = {set b(v) {console.log('setter called')}, get b() {return 1}}
a.b = a.b || 1 // will call setter
a.b ||= 1 // will not call setter
```

I thought this is what you mean, if not, please ignore.

**Comment by Rec0iL99** [comment]
@eslint/eslint-team what are the next steps here?

**Comment by nzakas** [comment]
> I think we can update the rule to report a = a || 1 || 2 and treat this change as a semver-minor bug fix. Curious what other team members think.

@mdjermanovic that makes sense to me. So you're thinking this is just a bug fix rather than adding a new option?

**Comment by mdjermanovic** [comment]
Marked accepted, PR is welcome. I think there's no need for a new option.
